src/sid/simulate.py

```diff
@@ -328,11 +331,10 @@ def _simulate(
     seed = np.random.randint(0, 1_000_000) if seed is None else seed
     seed = itertools.count(seed)
 
-    cum_probs = _prepare_assortative_matching_probabilities(
+    assortative_matching_probabilities = _prepare_assortative_matching_probabilities(
```

Comment:
we should keep the information that this is already the `cumsum` of the probabilities. Otherwise, of course the new name is better.